pkg/azuredisk/controllerserver_v2.go

```diff
@@ -488,7 +532,9 @@ func (d *DriverV2) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest)
 			return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%d) can not be negative", start)
 		}
 	}
-	if d.cloud.KubeClient != nil && d.cloud.KubeClient.CoreV1() != nil && d.cloud.KubeClient.CoreV1().PersistentVolumes() != nil {
+
+	kubeClient := d.controllerProvisioner.GetKubeClient()
+	if kubeClient != nil && kubeClient.CoreV1() != nil && kubeClient.CoreV1().PersistentVolumes() != nil {
```

Comment:
Since the K8s client is still needed by the server components, it makes sense for it to create one and pass it to the `provisioner.New()` method.